tools/analyzers/recursivelock/testdata/complexlocks.go

```diff
@@ -0,0 +1,23 @@
+// These nested rlock patterns are too complex for the analyzer to catch right now
```

Comment:
Given that there are certain classes of locks that this analyzer won't catch, do you mind opening an issue to track them ? This way if anyone wants to solve it and needs context they can fall back on the issue.

Reply:
Sure! I'll make it right now.

Reply:
#10078